File: c++/binarysearch.cpp
#include <iostream>
#include <vector>
using namespace std;

int binarysearch(vector<int> &arr,int x) {
    int low = 0;
    int high = arr.size() -1 ;

    while(low<=high) {
        int mid = low + (high - low) / 2;
        if (arr[mid] == x){
            return mid;
        }
        else if (arr[mid]<x){
            low = mid + 1;
        }

        else 
            high = mid -1 ;
        
    }
    return -1;

}


int main() {
    vector<int> arr = {1, 3, 5, 7, 9, 11};
    int x = 7;

    int idx = binarysearch(arr, x);
    if (idx != -1) {
        cout << "Element found at index " << idx << endl;
    } else {
        cout << "Element not found" << endl;
    }
}

File: algos/src/binarysearch.rs
pub fn binarysearch(arr:&[i32],x:i32) -> Option<usize> {
    let mut low = 0;
    let mut high = arr.len().wrapping_sub(1);
    while low<=high  {
    let mid = low + (high - low) / 2;
    if arr[mid]==x {
        return Some(mid);
    }

    else if arr[mid]<x {
        low = mid + 1;
    }
    else {
      
        high = mid - 1 ;
    }
    }
    None
}


